libs/agno/agno/agent/agent.py

```diff
@@ -3679,11 +3744,78 @@ def get_tools(
 
         return agent_tools
 
+    def _collect_joint_images(
+        self, 
+        run_messages: RunMessages,  # Pass the whole RunMessages instead of individual fields
+        run_response: RunOutput
+    ) -> Optional[Sequence[Image]]:
+        """Collect images from input, session history, and current run response."""
+        from agno.utils.log import log_debug
+        
+        joint_images = []
+        
+        # 1. Add images from current input (from RunMessages)
+        if run_messages.images:
+            joint_images.extend(run_messages.images)
+            log_debug(f"Added {len(run_messages.images)} input images to joint list")
+        
+        # 2. Add images from session history
+        try:
+            session = self.get_session()
+            if session and session.runs:
+                for historical_run in session.runs:
+                    if historical_run.run_id != run_response.run_id and historical_run.images:
```

Comment:
Is this possible though? You current run hasn't been stored yet. Seems like an unnecessary check, but maybe future proof

Reply:
No so this for a case like this-
1. run 1 -> image input direct to Agent with send_media_to_model=False so that its available only on the tool ImageAnalysisTool (so run 1 we have only 1 image)
2. run 2 -> no image input to agent in run, but dalle tool generates an image so now two images should be stored in `join_images` right
3. run 3 -> now here we should have access to both images in the ImageAnalysisTool

If you run the cookbook you'll see it. or am i getting something wrong?

Reply:
ah true